src/algebra/module/zlattice.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2023 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import analysis.normed_space.basic
+import measure_theory.group.fundamental_domain
+
+/-!
+# ℤ-lattices
+Let `E` be a finite dimensional real vector space. A (full) ℤ-lattice `L` of `E` is a discrete
+subgroup of `E` such that `L` spans `E` over `ℝ`.
+
+The ℤ-lattice `L` can be defined in two ways:
+* For `b` a basis of `E`, then `L : submodule.span ℤ (set.range b)` is a ℤ-lattice of `E`.
+* As `L : add_subgroup E` with the additional properties:
+  `(hd : ∀ r : ℝ, (L ∩ (metric.closed_ball 0 r)).finite)`, that is `L` is discrete
+  `(hs : submodule.span ℝ (L : set E) = ⊤)`, that is `L` spans `E` over `ℝ`.
+
+## Main results
+* `zspan.is_add_fundamental_domain`: proves that the set defined by `zspan.fundamental_domain` is
+indeed a fundamental domain of the lattice.
+* `zlattice.dim`: for `L : add_subgroup E` with `L` discrete and spanning `E` over `ℝ`, proves that
```

Comment:
I can't see this result, where is it? in a dependent PR I guess?

Reply:
Yes, it's in #18477. Somehow, the docstring leaked into this PR...